// @sanity/visual-editing/dist/overlay-components/index.cjs
"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("../_chunks-cjs/PointerEvents.cjs"),t=require("react/jsx-runtime"),o=require("react"),n=require("../_chunks-cjs/mutations.cjs");const a=e.d(e.B)`
  position: relative;
  transform: var(--add-button-position);

  --add-button-position: translateY(0);
  [data-position='top'] & {
    --add-button-position: translateY(-50%);
  }
  [data-position='right'] & {
    --add-button-position: translateX(50%);
  }
  [data-position='bottom'] & {
    --add-button-position: translateY(50%);
  }
  [data-position='left'] & {
    --add-button-position: translateX(-50%);
  }
`,i=e.d(e.F)`
  pointer-events: all;
  height: var(--hover-area-height);
  width: var(--hover-area-width);

  --hover-area-height: 100%;
  --hover-area-width: 100%;
  &[data-position='top'],
  &[data-position='bottom'] {
    --hover-area-height: 48px;
  }
  &[data-position='right'],
  &[data-position='left'] {
    --hover-area-width: 48px;
  }
`,r=n=>{const{element:r,hoverAreaExtent:s,node:l,onAddUnion:d,position:u}=n,[c,p]=o.useState(!1),h=o.useCallback((()=>{p(!0)}),[]),v=o.useCallback((()=>{p(!1)}),[]),f=o.useRef(null),b=e.u(),m=o.useCallback((e=>{if(e.target===f.current){const t=new MouseEvent(e.type,{...e.nativeEvent,bubbles:!0,cancelable:!0});r.dispatchEvent(t)}}),[r]),[x,g]=o.useState(null),[j,E]=o.useState(!1),C=o.useCallback((()=>{E(!1),p(!1)}),[]),k=o.useCallback((e=>{E(!1),d("top"===u||"left"===u?"before":"after",e.name),b("Visual Editing Insert Menu Item Inserted",null)}),[d,u,b]);/* @__PURE__ */
return t.jsxs(i,{align:"top"===u?"flex-start":"bottom"===u?"flex-end":"center","data-position":u,"data-sanity-overlay-element":!0,justify:"left"===u?"flex-start":"right"===u?"flex-end":"center",onClick:m,onContextMenu:m,onMouseDown:m,onMouseEnter:h,onMouseLeave:v,onMouseUp:m,ref:f,style:{["top"===u||"bottom"===u?"height":"width"]:s},children:[(c||j)&&/* @__PURE__ */t.jsx(a,{ref:g,icon:e.A,mode:"ghost",onClick:()=>E((e=>!e)),radius:"full",selected:j}),j&&x&&/* @__PURE__ */t.jsx(e.I,{node:l,onDismiss:C,referenceElement:x,onSelect:k})]})};exports.PointerEvents=e.P,exports.UnionInsertMenuOverlay=a=>{const{direction:i="vertical",element:s,hoverAreaExtent:l,node:d,parent:u}=a,{getDocument:c}=n.u(),p=o.useCallback(((e,t)=>{const o=c(d.id),a=n.a(d,t,e);o.patch(a)}),[c,d]);return u?/* @__PURE__ */t.jsxs(e.F,{height:"fill",width:"fill",direction:"horizontal"===i?"row":"column",justify:"space-between",children:[
/* @__PURE__ */t.jsx(r,{element:s,hoverAreaExtent:l,node:u,onAddUnion:p,position:"horizontal"===i?"left":"top"}),
/* @__PURE__ */t.jsx(r,{element:s,hoverAreaExtent:l,node:u,onAddUnion:p,position:"horizontal"===i?"right":"bottom"})]}):null},exports.defineOverlayComponent=function(e,t){return{component:e,props:t}},exports.defineOverlayComponents=function(e){return e};//# sourceMappingURL=index.cjs.map
